include/Surelog/Common/FileSystem.h

```diff
@@ -0,0 +1,70 @@
+/*
+ Copyright 2022 chipsalliance
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+/*
+ * File:   FileSystem.cpp
+ * Author: hs
+ *
+ * Created on June 1, 2022, 3:00 AM
+ */
+
+#ifndef SURELOG_FILESYSTEM_H
+#define SURELOG_FILESYSTEM_H
+#pragma once
+
+#include <cstdint>
+#include <filesystem>
+#include <string_view>
+
+namespace SURELOG {
+class PathId;
+class SymbolId;
+class SymbolTable;
+
+class FileSystem {
+ public:
+  static FileSystem *getInstance();
+  static FileSystem *setInstance(FileSystem *const fileSystem);
+
+ public:
+  virtual PathId toPathId(const std::filesystem::path &path,
+                          SymbolTable *const symbolTable);
+  virtual const std::string &toSymbol(PathId id);
+  virtual std::filesystem::path toPath(PathId id);
+
+  virtual const std::filesystem::path &getCwd();
+  virtual PathId getCwd(SymbolTable *const symbolTable);
+
+  virtual PathId copy(PathId id, SymbolTable *const toSymbolTable);
```

Comment:
Given that this is a class called `FileSystem`, a function called `copy()` copies files around, right ?

If not, this needs a different name, and also, it needs a description.

Also, don't make the parameter `const`, it makes absolutely no difference (as you `const` the value, not the point location), and thus is just confusing.

Reply:
In which contexts is this used ? Maybe `duplicatePath()` could be a good name ?
What are the scenarios where this will be used ?